src/webgpu/shader/execution/expression/expression.ts

```diff
@@ -969,6 +1019,53 @@ export function generateVectorPairToVectorCases(
   }, new Array<Case>());
 }
 
+/**
+ * @returns a Case for the param and vector of intervals generator provided
+ * @param param the param to pass in
+ * @param filter what interval filtering to apply
+ * @param ops callbacks that implement generating an vector of acceptance
+ *            intervals for a vector.
+ */
+function makeMatrixToMatrixCase(
+  param: number[][],
+  filter: IntervalFilter,
+  ...ops: MatrixToMatrix[]
+): Case | undefined {
+  param = map2DArray(param, quantizeToF32);
+  const param_f32 = map2DArray(param, f32);
+
+  const results = ops.map(o => o(param));
+  if (filter === 'f32-only' && results.some(m => m.some(c => c.some(r => !r.isFinite())))) {
```

Comment:
This should be at most 16 checks per result/op, so not unboundedly expensive, but I agree that this could be improved with additional data.

I think that would require a bit of thinking/design to pick the right way to do that. In theory Vector has a similar issue. I think I would prefer to leave as is in this patch, and file a follow up bug to investigate optimizing this check, if that is ok with you.

Reply:
Yes, we can follow up with optimziations.